### ISSUE TITLE
[curlpp] build failure

### ISSUE THREAD

**Issue by cenit** [microsoft/vcpkg#10714]
**Host Environment**
- OS: Windows
- Compiler: VS 2019 16.5.2

**To Reproduce**
`./vcpkg install curlpp`

**Failure logs** 
```
-- Found CURL version: 7.68.0-DEV
-- Using CURL include dir(s): C:/vcpkg/installed/x86-windows/include
-- Using CURL lib(s): optimized;C:/vcpkg/installed/x86-windows/lib/libcurl.lib;debug;C:/vcpkg/installed/x86-windows/debug/lib/libcurl-d.lib;wldap32;winmm;ws2_32;advapi32;crypt32;advapi32;ZLIB::ZLIB
-- Configuring done
CMake Error at C:/vcpkg/scripts/buildsystems/vcpkg.cmake:215 (_add_library):
  Target "curlpp" links to target "ZLIB::ZLIB" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:89 (add_library)
```

**Additional context**
The problem is *not* related to curlpp itself, but to curl's `vcpkg-cmake-wrapper.cmake`, which is missing a key `find_package(ZLIB)`

**Comment by cenit** [comment]
Note that curlConfig contains a find_dependency(ZLIB). Somehow it is not exported downstream.
Might be due to a combination of our vcpkg-cmake-wrapper and the new CMake 3.17. I do not remember any problem before installing curlpp, and nothing has changed apart from cmake in the meantime

**Comment by timautry** [comment]
I tried .\vcpkg.exe install curlpp and I had the following errors in the config-x64-windows-out.log about missing OpenSSL and ZLIB:

CMake Error at D:/Git/vs2019.prod/scripts/buildsystems/vcpkg.cmake:215 (_add_library):
  Target "curlpp" links to target "OpenSSL::SSL" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:89 (add_library)


CMake Error at D:/Git/vs2019.prod/scripts/buildsystems/vcpkg.cmake:215 (_add_library):
  Target "curlpp" links to target "OpenSSL::Crypto" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:89 (add_library)


CMake Error at D:/Git/vs2019.prod/scripts/buildsystems/vcpkg.cmake:215 (_add_library):
  Target "curlpp" links to target "ZLIB::ZLIB" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:89 (add_library)

**Comment by cenit** [comment]
I had only the last because I have CURL without SSL feature. Might be that the wrapper is missing the find_package call for every dependency...

**Comment by cenit** [comment]
@timautry thanks for your test. Do you have CMake 3.17?
I will try to install CURL with all features and see what happens

**Comment by timautry** [comment]
@cenit, yes I have Cmake 3.17 installed, latest of VS 2017 and 2019 and both fail with the OpenSSL and ZLIB missing.

**Comment by cenit** [comment]
ok. But I was already almost sure, since I can see it failing in all CI configuration that use the latest VS and CMake

**Comment by timautry** [comment]
You are most welcome! 
Also I have noticed some ports (cpuid, metrohash, murmurhash, stormlib) require the portfile.cmake to include the line:
**set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)**
before they will install properly as it states **The folder /include is empty or not present. This indicates the library was not correctly installed**.
Seems this started with CMake version 3.16.5 and continues into 3.17.0.
I opened an earlier ticket **[libcpuid and stormlib] build failure x64-windows /include is empty or not present #10383**, but it could not duplicated, most likely because they may be using 13.4 vcpkg has in downloads/tools or 3.12/3.16.0 installed with visual studio 2017/2019.
